src/components/pipelines-tasks/ClusterTaskRow.tsx

```diff
@@ -0,0 +1,170 @@
+import { ClusterTaskModel } from '../../models';
+import { useTranslation } from 'react-i18next';
+import {
+  KEBAB_ACTION_DELETE_ID,
+  KEBAB_ACTION_EDIT_ANNOTATIONS_ID,
+  KEBAB_ACTION_EDIT_ID,
+  KEBAB_ACTION_EDIT_LABELS_ID,
+  KEBAB_BUTTON_ID,
+  TaskGroupVersionKind,
+} from '../../consts';
+import { useHistory } from 'react-router-dom';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownPosition,
+  KebabToggle,
+} from '@patternfly/react-core';
+
+type ClusterTaskKebabProps = {
+  obj: K8sResourceCommon;
+};
+
+const ClusterTaskKebab: React.FC<ClusterTaskKebabProps> = ({ obj }) => {
+  const { t } = useTranslation('plugin__pipeline-console-plugin');
+  const [isOpen, setIsOpen] = React.useState(false);
+  const launchDeleteModal = useDeleteModal(obj);
+  const launchAnnotationsModal = useAnnotationsModal(obj);
+  const launchLabelsModal = useLabelsModal(obj);
+  const { name, namespace } = obj.metadata;
+  const canEditCronTab = useAccessReview({
+    group: TaskGroupVersionKind.group,
+    resource: ClusterTaskModel.plural,
+    verb: 'update',
+    name,
+    namespace,
+  });
+  const canDeleteCronTab = useAccessReview({
+    group: TaskGroupVersionKind.group,
+    resource: ClusterTaskModel.plural,
+    verb: 'delete',
+    name,
+    namespace,
+  });
+  const history = useHistory();
+
+  const onToggle = (isOpen: boolean) => {
+    setIsOpen(isOpen);
+  };
+
+  const onFocus = () => {
+    const element = document.getElementById('kebab-button');
+    element.focus();
+  };
+
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  const editURL = `/k8s/ns/${namespace}/${getReferenceForModel(
+    ClusterTaskModel,
+  )}/${encodeURIComponent(name)}/yaml`;
+
+  const dropdownItems = [
+    <DropdownItem
+      key={KEBAB_ACTION_EDIT_LABELS_ID}
+      component="button"
+      onClick={launchLabelsModal}
+      isDisabled={!canEditCronTab}
+      data-test-action={KEBAB_ACTION_EDIT_LABELS_ID}
+    >
+      {t('Edit labels')}
+    </DropdownItem>,
+    <DropdownItem
+      key={KEBAB_ACTION_EDIT_ANNOTATIONS_ID}
+      component="button"
+      onClick={() => launchAnnotationsModal()}
+      isDisabled={!canEditCronTab[0]}
+      data-test-action={KEBAB_ACTION_EDIT_ANNOTATIONS_ID}
+    >
+      {t('Edit annotations')}
+    </DropdownItem>,
+    <DropdownItem
+      key={KEBAB_ACTION_EDIT_ID}
+      component="button"
+      onClick={() => history.push(editURL)}
+      isDisabled={!canEditCronTab[0]}
+      data-test-action={KEBAB_ACTION_EDIT_ID}
+    >
+      {t('Edit {{resourceKind}}', { resourceKind: ClusterTaskModel.kind })}
+    </DropdownItem>,
+    <DropdownItem
+      key={KEBAB_ACTION_DELETE_ID}
+      component="button"
+      onClick={launchDeleteModal}
+      isDisabled={!canDeleteCronTab[0]}
+      data-test-action={KEBAB_ACTION_DELETE_ID}
+    >
+      {t('Delete {{resourceKind}}', { resourceKind: ClusterTaskModel.kind })}
+    </DropdownItem>,
+  ];
```

Comment:
Can you create a common kebab component called `K8sCommonKebabMenu` and use it? This component will have Edit annotation, Edit labels, Edit Resource(which will redirect to yaml page of the resource), and Delete options.